pkg/syndesis/reconcile.go

```diff
@@ -1,10 +1,37 @@
 package syndesis
 
-import api "github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+import (
+	api "github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/action"
+)
+
+var (
+	actionPool	[]action.InstallationAction
+)
+
+func init() {
+	actionPool = append(actionPool,
+		&action.Initialize{},
+		&action.Install{},
```

Comment:
I wonder whether we  need to distinguish those two phases or whether we shouldn't combine initialize and install ? Before doing the install we could do the sanity check that only a single `Syndesis` is installed.

Reply:
So, I've added the `initialize` stage in order to have a initial step to execute as soon as the CR is created. I was thinking that the next state may be "Installing" or "Upgrading" or "Reconciling", depending on what we'll find out when we'll work on "upgrade from operator" or "upgrade from a previous template installation". We may remove it if it will not be useful.

Putting a state diagram in the readme may be useful..